src/Schema/Comparator.php

```diff
@@ -44,6 +46,32 @@ public function __construct(?AbstractPlatform $platform = null)
         $this->platform = $platform;
     }
 
+    /**
+     * @param array{Schema, Schema} $args
```

Comment:
This type looks odd. It assumes that no other methods will be called magically. I believe it could be loosened to something like `list<mixed>` (I wouldn't care about the named arguments).

Reply:
If I do that, static analysis is not going to be happy about line 58 (`Parameter #1 $fromSchema of method Doctrine\DBAL\Schema\Comparator::doCompareSchemas() expects Doctrine\DBAL\Schema\Schema, mixed given.`). Should I then assert the type of `$args[0]` and `$args[1]`?

Reply:
Before invoking the private method, the magic one should validate the arguments accordingly.

Reply:
No, just silence that error. `__call()` and `__callStatic()` will catch any calls to unknown methods, so we can't really make any assumptions on the types of the arguments. And the error that PHP raises if the arguments are of the wrong type is actually desired.

So imho, all we need to do here is to tell the static analyzers that we know what we're doing.

Reply:
Ok, let's do that.

Reply:
Just figured out that I get an error in my IDE because I'm forcing phpstan to use level max. There is no issue in the CI actually :)